src/tokenizedSubaccounts/PPTSA.sol

```diff
@@ -0,0 +1,492 @@
+    PPTSAStorage storage $ = _getPPTSAStorage();
+    uint strikeDiff = higherStrike.strike - lowerStrike.strike;
+    if (strikeDiff != _getPPTSAStorage().ppParams.strikeDiff) {
+      revert PPT_StrikePriceOutsideOfDiff();
+    }
+
+    if (higherStrike.tradeAmount != -lowerStrike.tradeAmount) {
+      revert PPT_InvalidTradeAmount();
+    }
+
+    (uint openSpreads, uint baseBalance,) = _getSubAccountStats();
+
+    _verifyFee(maxFee, _getBasePrice());
+    _validateTradeDetails(lowerStrike, higherStrike);
+
+    uint maxLossOfOpenOptions = openSpreads.multiplyDecimal(strikeDiff);
+    uint totalTradeMaxLoss = higherStrike.tradeAmount.abs().multiplyDecimal(strikeDiff);
+
+    if (
+      maxLossOfOpenOptions + totalTradeMaxLoss
+        > baseBalance.multiplyDecimal(_getBasePrice()).multiplyDecimal($.ppParams.maxBuyPctOfTVL)
+    ) {
+      revert PPT_TradeTooLarge();
+    }
+  }
+
+  /////////////////
+  // Option Math //
+  /////////////////
+
+  function _validateTradeDetails(StrikeData memory lowerStrike, StrikeData memory higherStrike) internal view {
+    PPTSAStorage storage $ = _getPPTSAStorage();
+    int markCost = lowerStrike.markPrice.multiplyDecimal(lowerStrike.tradeAmount)
+      + higherStrike.markPrice.multiplyDecimal(higherStrike.tradeAmount);
+
+    int totalCostOfTrade = lowerStrike.tradePrice.toInt256().multiplyDecimal(lowerStrike.tradeAmount)
+      + higherStrike.tradePrice.toInt256().multiplyDecimal(higherStrike.tradeAmount);
+
+    if ($.ppParams.isLongSpread) {
+      if ($.ppParams.maxTotalCostTolerance < 1e18) {
+        revert PPT_InvalidCostTolerance();
+      } else if (totalCostOfTrade.abs() > markCost.abs().multiplyDecimal($.ppParams.maxTotalCostTolerance)) {
+        revert PPT_TotalCostOverTolerance();
+      }
+    } else {
+      if ($.ppParams.maxTotalCostTolerance > 1e18) {
+        revert PPT_InvalidCostTolerance();
+      } else if (totalCostOfTrade.abs() < markCost.abs().multiplyDecimal($.ppParams.maxTotalCostTolerance)) {
+        revert PPT_TotalCostBelowTolerance();
+      }
+    }
+
+    uint markValueToStrikeDiffRatio =
+      ((lowerStrike.markPrice - higherStrike.markPrice).abs().divideDecimal(higherStrike.strike - lowerStrike.strike));
+
+    if (
+      markValueToStrikeDiffRatio < $.ppParams.minMarkValueToStrikeDiffRatio
+        || markValueToStrikeDiffRatio > $.ppParams.maxMarkValueToStrikeDiffRatio
+    ) {
+      revert PPT_MarkValueNotWithinBounds();
+    }
+  }
+
+  function _getMarkPrice(IRfqModule.TradeData memory trade)
+    internal
+    view
+    returns (uint expiry, uint strike, uint markPrice)
+  {
+    PPTSAStorage storage $ = _getPPTSAStorage();
+    (uint optionExpiry, uint optionStrike, bool isCall) = OptionEncoding.fromSubId(trade.subId.toUint96());
+    if ($.ppParams.isCallSpread != isCall) {
+      revert PPT_WrongInputSpread();
+    }
+    if (block.timestamp >= optionExpiry) {
+      revert PPT_OptionExpired();
```

Comment:
by switching the _getOptionPrice check from:
```
uint timeToExpiry = optionExpiry - block.timestamp;
if (timeToExpiry < params.optionMinTimeToExpiry || timeToExpiry > params.optionMaxTimeToExpiry) {
   revert BCMTSA_OptionExpiryOutOfBounds();
}
```
    
to:
```
if (expiry < block.timestamp + params.optionMinTimeToExpiry || expiry > block.timestamp + params.optionMaxTimeToExpiry) {
   revert BCMTSA_OptionExpiryOutOfBounds();
}
```

you can avoid adding this extra check here (since you don't have the concerns around overflows anymore)

Reply:
Nice call. Done